### ISSUE TITLE
1.0 architecture considerations

### ISSUE THREAD

**Issue by daffl** [feathersjs/feathers-authentication#226]
Authentication is a complex topic and finding a good balance between ease of use and flexibility can be tricky sometimes. For version 1.0 of this plugin we probably should try and reduce or split up the amount of code and API and configuration option surface area. Here are some of my thoughts on this:
- Should the plugin still serve [static failure and success pages](https://github.com/feathersjs/feathers-authentication/tree/master/src/public)? It seems to me like you'll never really need them.
- Should we try and split different authentication mechanism in their own modules. That would help to reduce the large number of configuration options and keep things modular. I was thinking of something like

``` js
const feathers = require('feathers');
const authentication = require('feathers-authentication');
const local = require('feathers-authentication-local');
const oauth2 = require('feathers-authentication-oauth2');

const app = feathers();

app.configure(authentication({
  secret: 'myauthsecret'
}));
app.configure(local({
  endpoint: 'auth/local'
});
app.configure(oauth2({
  facebook: {
    strategy: FacebookStrategy,
    'clientID': 'your facebook client id',
    'clientSecret': 'your facebook client secret',
    'permissions': {
      authType: 'rerequest',
      'scope': ['public_profile', 'email']
    }
  }
}));
```
- Add the ability to `remove` a token. I wrote some thoughts on that up at https://github.com/feathersjs/feathers-authentication/issues/73#issuecomment-227008676

**Comment by ekryski** [comment]
Yes I agree. I think if we can split it out that would be great. It will reduce the dependencies for `feathers-authentication`. I'd like to get close to 1.0 (ie. password reset, passwordless auth) first before we refactor but I had been thinking something similar.

**Comment by ekryski** [comment]
And I think we should pull the static pages. You and I discussed over hangout but they really just get in the way and almost 100% of the time end up getting overridden.

**Comment by ekryski** [comment]
This is pretty much complete in the [passwordless](https://github.com/feathersjs/feathers-authentication/tree/passwordless) branch.

**Comment by ekryski** [comment]
I think we can close this out now. We have had a lot of discussion and iteration over the last month and have settled on an adapter for Passport. It's working really well and is now in master and released slated to go out as 1.0.0.

If you want to see more detail you can check the code, examples, and docs.
